artemis-server/src/main/java/org/apache/activemq/artemis/core/deployers/impl/FileConfigurationParser.java

```diff
@@ -253,9 +253,12 @@ public Configuration parseMainConfig(final InputStream input) throws Exception {
       String xml = XMLUtil.readerToString(reader);
       xml = XMLUtil.replaceSystemProps(xml);
       Element e = XMLUtil.stringToElement(xml);
-
+      NodeList children = e.getElementsByTagName("core");
```

Comment:
Why not just get the broker.xml fully validated by the xsd using javax.xml.validation.Validator instead of manually getting elements and recursing.

Reply:
e.g.
      SchemaFactory schemaFactory = SchemaFactory.newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI);
      Schema schema = schemaFactory.newSchema(new File("schema/artemis-configuration.xsd"));
      Validator validator = schema.newValidator();
      validator.validate(new DOMSource(e));

Reply:
Yes, that seems better than my solution. I wanted to use the same process while reloading conf. as parsing conf. when starting broker (https://github.com/apache/activemq-artemis/blob/master/artemis-server/src/main/java/org/apache/activemq/artemis/core/config/FileDeploymentManager.java#L85).

Reply:
Hmm, after another investigation I am not sure if this is possible with current XSD schema. It does not contain 'configuration' element, so it cannot be checked this way. There are two solutions - modify XSD schema to contain missing element or use my solution, where I am getting first child of 'configuration', which is 'core'.

Reply:
https://github.com/apache/activemq-artemis/blob/master/artemis-server/src/main/resources/schema/artemis-server.xsd it does.

Reply:
Ah, didn't know about this, sorry. Should be ok now.